src/remoteCollection/handleCollectResponse.js

```diff
@@ -62,21 +80,66 @@ function handleCollectResponse(collectResponse) {
   return collectResponse.then((collectRes) => {
     try {
       validateCollectResponse(collectRes);
+
+      /*
+       * If the transform is a string, then we use that function for all
+       * status codes.
+       */
       const tr = collectRes.generatorTemplate.transform;
-      const t = Array.isArray(tr) ? tr.join('\n') : tr;
-      const transformedSamples = evalUtils.safeTransform(t, collectRes);
-      logger.info(`{
-        generator: ${collectRes.name},
-        numSamples: ${transformedSamples.length},
-      }`);
-      enqueue(transformedSamples);
+      if (typeof tr === 'string') { // match all status codes
+        const samplesToEnqueue = evalUtils.safeTransform(tr, collectRes);
+        logger.info({
+          generator: collectRes.name,
+          url: collectRes.url,
+          numSamples: samplesToEnqueue.length,
+        });
+        return enqueue(samplesToEnqueue);
+      }
+
+      /*
+       * The transform is *not* a string, so check for a status code regex
+       * match which maps to a transform for error samples. Use the first one
+       * to match.
+       */
+      Object.keys(tr).forEach((statusMatcher) => {
```

Comment:
Just one thing I need to confirm,
If transform is not a string, we will always have a transform matching status 200 (and possibly other overrided status matchings)?

Reply:
> always have a transform matching status 200
No, the transform developer would specify a RegExp as a key and a transform function as the value. If the RegExp key matches the status code, then use that value as the transform fn.

e.g. here the SGT developer specifies a function to handle 200 and 3xx in one way, and 404 another way, but wants to just generate standard "default" error samples for any other status codes:

"transform": {
  "(200)|(3\d\d)": "const samples = []; /* do status OK stuff here */ return samples;",
  "404": "const samples = []; /* do NOT_FOUND stuff here */ return samples;"
}

Reply:
From what I understand, the transform is a code that is executed by the eval function. Won't  this transform throw an error when executed?

Reply:
if transform is a string, we just execute it in a vm sandbox but when it's an object, we check the key for a regexp match to response's http status code and execute the associated value when the status matches

Reply:
Ok, I see what you are saying. It is like, specifying a different transform for a set of error code.